packages/backend/src/auth/auth.service.ts

```diff
@@ -72,58 +73,74 @@ export const signup = async (req: express.Request, res: express.Response) => {
   }
 }
 
-export const login = async (req: express.Request, res: express.Response) => {
-  const { email, password } = await zParse(loginSchema, req)
+export const login = async (
+  req: express.Request,
+  res: express.Response,
+  next: NextFunction
+) => {
+  try {
+    const { email, password } = await zParse(loginSchema, req)
 
-  const user = await User.query()
-    .findOne({ email })
-    .select('id', 'email', 'password')
-    .throwIfNotFound(new UnauthorisedException('Invalid credentials'))
-    .withGraphFetched('refreshTokens')
+    const user = await User.query()
+      .findOne({ email })
+      .select('id', 'email', 'password')
+      .withGraphFetched('refreshTokens')
 
-  const isValid = await user.verifyPassword(password)
-  if (!isValid) {
-    throw new UnauthorisedException('Invalid credentials')
-  }
+    if (!user) {
+      throw new UnauthorisedException('Invalid credentials')
+    }
 
-  let refreshToken = user.refreshTokens?.[0]
+    const isValid = await user.verifyPassword(password)
+    if (!isValid) {
+      throw new UnauthorisedException('Invalid credentials')
+    }
 
-  const { refreshToken: generatedToken, expiresIn: refreshTokenExpiresIn } =
-    generateRefreshToken(user.id)
+    let refreshToken = user.refreshTokens?.[0]
 
-  if (refreshToken) {
-    refreshToken.token = generatedToken
-    refreshToken.expiresAt = new Date(Date.now() + refreshTokenExpiresIn * 1000)
+    const { refreshToken: generatedToken, expiresIn: refreshTokenExpiresIn } =
+      generateRefreshToken(user.id)
 
-    await refreshToken.$query().patch(refreshToken)
-  } else {
-    refreshToken = new RefreshToken()
-    refreshToken.token = generatedToken
-    refreshToken.expiresAt = new Date(Date.now() + refreshTokenExpiresIn * 1000)
+    if (refreshToken) {
+      refreshToken.token = generatedToken
+      refreshToken.expiresAt = new Date(
+        Date.now() + refreshTokenExpiresIn * 1000
+      )
 
-    await user.$relatedQuery('refreshTokens').insert(refreshToken)
-  }
+      await refreshToken.$query().patch(refreshToken)
+    } else {
+      refreshToken = new RefreshToken()
+      refreshToken.token = generatedToken
+      refreshToken.expiresAt = new Date(
+        Date.now() + refreshTokenExpiresIn * 1000
+      )
+
+      await user.$relatedQuery('refreshTokens').insert(refreshToken)
+    }
 
-  const { accessToken, expiresIn: accessTokenExpiresIn } = generateJWT(user.id)
+    const { accessToken, expiresIn: accessTokenExpiresIn } = generateJWT(
+      user.id
+    )
 
-  appendTokensToCookie(
-    res,
-    accessToken,
-    accessTokenExpiresIn,
-    refreshToken.token,
-    refreshTokenExpiresIn
-  )
+    appendTokensToCookie(
+      res,
+      accessToken,
+      accessTokenExpiresIn,
+      refreshToken.token,
+      refreshTokenExpiresIn
+    )
 
-  return res.send({ user })
+    return res.send({ user })
```

Comment:
This should be returned as a `Response`:
https://github.com/interledger/testnet/blob/38f401b207a0bd8f6778bc65b5fbd50883310fc2/packages/backend/src/shared/models/Response.ts#L1

I'll probably rename the interface since there already two `Response` types: from `node` and `express`